packages/@lwc/synthetic-shadow/src/faux-shadow/events.ts

```diff
@@ -113,6 +117,9 @@ function getWrappedShadowRootListener(listener: EventListener): WrappedListener
             if (shouldInvoke) {
                 listener.call(currentTarget, event);
             }
+
+            // Disable retargeting for this event instance (see Event.prototype.target polyfill).
```

Comment:
Should wrap the invocation of the original listener in a try-finally and disable retargeting in the finally block. This handles the cases where the original listener throws an error and the event still continues to propagate.

Reply:
I thought about this but I believe the event will stop propagating when an exception is thrown because the event lifecycle is sync. I'll add a test for it to verify.

Reply:
Verified that we do need the try-finally. It seems that all queued listeners execute before program termination.